viamonvif/discovery.go

```diff
@@ -274,31 +281,54 @@ func GetCameraInfo(
 	return cameraInfo, nil
 }
 
-// GetRTSPStreamURIsFromProfiles uses the ONVIF Media service to get the RTSP stream URLs for all available profiles.
-func GetRTSPStreamURIsFromProfiles(
+// GetMediaInfoFromProfiles uses the ONVIF Media service to get the RTSP stream URLs
+// and Snapshot URIs for all available profiles.
+func GetMediaInfoFromProfiles(
 	ctx context.Context,
 	dev OnvifDevice,
 	creds device.Credentials,
 	logger logging.Logger,
-) ([]string, error) {
+) ([]MediaInfo, error) {
 	resp, err := dev.GetProfiles(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// Resultant slice of RTSP URIs
-	var rtspUris []string
-
+	var mes []MediaInfo
 	// Iterate over all profiles and get the RTSP stream URI for each one
 	for _, profile := range resp.Profiles {
-		uri, err := dev.GetStreamURI(ctx, profile, creds)
+		logger.Debugf("Looking up media info for profile: %s (token: %s)", profile.Name, profile.Token)
+		streamURI, err := dev.GetStreamURI(ctx, profile.Token, creds)
+		if err != nil {
+			logger.Warn(err.Error())
+			continue
+		}
+
+		// Check if the stream URI is empty
+		if streamURI == nil || streamURI.String() == "" {
+			logger.Warnf("Stream URI is empty for profile %s", profile.Name)
+			continue
+		}
+
+		snapshotURI, err := dev.GetSnapshotURI(ctx, profile.Token, creds)
 		if err != nil {
```

Comment:
If `GetSnapshotURI` fails, we should still add the successful GetStreamURI response to the returned MediaInfo

Why would we ever want to continue here?

Reply:
Good call, thanks. Removed `continue` if GetSnapshotURI fails -- we now still return MediaInfo for the profile stream URIs.